doc/charm++/manual.rst

```diff
@@ -2555,173 +2557,196 @@ Examples are in ``examples/charm++/load_balancing`` and
+detail, along with configuration and execution instructions. The
+following strategies can be used with TreeLB (the old runtime
+selection syntax still works and is specified in parentheses):
+
+- **Greedy**: Uses a greedy algorithm that iterates over the objects
+  and assigns the heaviest remaining object to the least loaded
+  processor. (Old: ``+balancer GreedyLB``)
+
+- **GreedyRefine**: Uses a greedy algorithm that assigns the heaviest
+  remaining object to the least loaded processor when it is currently
+  assigned to a heavily loaded processor, otherwise leaves the object
+  on its current processor to limit migrations. It takes an optional
+  argument ``tolerance``, which specifies the tolerance it should allow
+  above the maximum load Greedy would produce. (Old: ``+balancer GreedyRefineLB``)
+
+- **RefineA, RefineB**: Moves objects away from the most overloaded
+  processors to reach average, limits the number of objects
+  migrated. RefineA allows a heavy object to go to any of the lightly
+  loaded PEs, while RefineB always moves the heaviest remaining object
+  to the lightest loaded PE. (Old: ``+balancer RefineLB`` runs RefineA)
+
+Listed below are load balancers intended for diagnostic purposes:
+
+- **Dummy**: Does nothing, does not move objects at all. (Old: ``+balancer DummyLB``)
+
+- **Random**: Randomly assigns objects to processors. (Old: ``+balancer RandCentLB``)
+
+- **Rotate**: Moves objects to the next available PE every time it is
+  called. It is useful for debugging PUP routines and other migration
+  related bugs. (Old: ``+balancer RotateLB``)
+
+However, not all non-distributed load balancers are currently
+supported by TreeLB, namely the centralized communication-aware load
+balancers:
+
+- **RecBipartLB**: Uses recursive bipartitioning to partition the
+  object communication graph.
+
+- **MetisLB**: Uses `METIS
+  <http://glaros.dtc.umn.edu/gkhome/metis/metis/overview>`__ to
+  partition the object communication graph. METIS is distributed with
+  Charm++, so there is no need to separately get this dependence.
+
+- **ScotchLB**: Uses the `SCOTCH
+  <http://www.labri.fr/perso/pelegrin/scotch/>`__ library for
+  partitioning the object communication graph, while also taking
+  object load imbalance into account. SCOTCH is not distributed with
+  Charm++, so end users must download and build the library from the
+  above link in order to use this load balancer. Because of this
+  dependence, ScotchLB is not built by default; it can be built by
+  running ``make ScotchLB`` in the Charm++ build folder
+  (e.g. ``netlrts-linux-x86_64-smp/``). If SCOTCH is installed in a
+  non-standard location, use the *-incdir* and *-libdir* build time
+  options to point to the include and library directories used,
+  respectively.
+
+In distributed approaches, load data is only exchanged among
+neighboring processors. There is no global synchronization. However,
```

Comment:
Ah, yes, it is perhaps a bit misleading. Global synchronization _with the application_ isn't needed for the centralized load balancers; once any object calls `AtSync()`, it can immediately continue executing, the load balancer will run concurrently whenever it is ready.

What this is trying to say is that the internals of the distributed load balancer itself does not make any globally synchronous calls, while TreeLB does (although it only does so on LB iterations that go up to the root level of the tree, in general it only requires synchronization up to the root of the largest subtree used for LB on that iteration).

Relaxing your global synchronization may buy you performance, potentially a significant amount. The main issues to consider are:

- PUP costs may be higher since the PUP routine can now run anytime, so it may have to transfer transient state. Additionally, it may put more strain on the network since regular application messages may also be in flight. In particularly high traffic scenarios, object migration can have the knock-on effect of creating additional messaging due to location manager misdelivering messages to stale locations and having to forward them.  
- The load balancing statistics may be less accurate. Stats are taken from the LB database when the LB actually starts running, not when `AtSync()` is called. Thus, if that point is not colocated with a global synchronization, different objects may have completed different amounts of work (and perhaps even be on different iterations, etc.), so the load reported can be noisy.
- In return for these costs, you avoid the idle time of waiting for LB to complete, potentially a big savings for large jobs, particularly expensive LB algorithms, and very imbalanced workloads.

Also, a minor point, but for the sake of completeness, calling an LB before the previous call has finished is undefined behavior as far as I know, so some synchronization at some point is technically needed, although in practice one could almost certainly completely forego synchronizing these calls with no ill effect for realistic applications.